src/main/java/org/elasticsearch/common/jna/Kernel32Library.java

```diff
@@ -37,17 +36,21 @@
 
     private static ESLogger logger = Loggers.getLogger(Kernel32Library.class);
 
-    private Kernel32 internal;
+    private boolean loaded = false;
```

Comment:
Ok, I understand your point. I think we can remove the boolean and the method isLoaded(). If the native lib cannot be loaded, the singleton is still instantiated and not null, but any call to a native method (or call to Kernel32Library.addConsoleCtrlHandler which calls the native SetConsoleCtrlHandler) will throw an UnsatisfiedLinkError which is completely acceptable to me since, well, we are doing native calls :)